### PR TITLE
[BEAM-2060] Allow to specify charset in XmlIO

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @lukecwik

File: sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java [hunk 1]
```diff
@@ -325,6 +355,13 @@ public boolean matches(String filename) {
       return toBuilder().setCompressionType(compressionType).build();
     }
 
+    /**
+     * Sets the XML file charset.
+     */
+    public Read<T> withCharset(String charset) {
```
Comment (dgouyette) [inline sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java]: using  java.nio.charset.Charset would be better, no ?

Comment (lukecwik) [inline sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java]: +1

Comment (jbonofre) [inline sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java]: `Charset` is not `Serializable`, that's why I preferred to use a simple `String`.

Comment (jbonofre) [inline sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java]: I'm updating to have the user facing API uses `withCharset(Charset)` and internally use the `String` representation of the `Charset` (for the serialization).

**Comment by dgouyette** [comment]
+ 1

File: sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java (hunk 1 above)
Comment (dgouyette) [inline sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java]: +1

**Comment by lukecwik** [comment]
LGTM (I'll merge this), do you want to do a follow up change so that a user can also specify the charset when writing as well?

**Comment by jbonofre** [comment]
Yes, let me add the write part in a new commit.

**Comment by lukecwik** [comment]
Great

**Comment by dhalperi** [comment]
My belief is that this was explicitly omitted because XMLIO is not actually correct with most charsets. How can you guard against bugs from users paramterizing with bad charsets here?

**Comment by lukecwik** [comment]
Only the javadoc comment that states that we don't support multi byte encodings.

**Pull request closed without merging**